src/steps/apply-transform.js

```diff
@@ -1,27 +1,31 @@
 import Resource from "../types/Resource";
 import Collection from "../types/Collection";
+import { Promise, isPromise } from "q";
 
 /**
  * @param toTransform Could be a single resource, a collection, a link object, or null.
  */
 export default function(toTransform, mode, registry, frameworkReq, frameworkRes) {
-  if(toTransform instanceof Resource) {
-    return transform(toTransform, frameworkReq, frameworkRes, mode, registry);
-  }
+  return Promise((resolve, reject) => {
 
-  else if (toTransform instanceof Collection) {
-    // below, allow the user to return undefined to remove a vlaue.
-    let newResources = toTransform.resources.map((it) =>
-      transform(it, frameworkReq, frameworkRes, mode, registry)
-    ).filter((it) => it !== undefined);
+    if(toTransform instanceof Resource) {
+      transform(toTransform, frameworkReq, frameworkRes, mode, registry).then(resolve, reject);
```

Comment:
Getting the `transform()` promise and using `.then` to resolve the outer promise feels really awkward. We can just do `return transform(toTransform, frameworkReq, frameworkRes, mode, registry)`, right?

Then, in the `else if` branch, I think we can just do `return Promise.all(....)`